src/OSPSuite.Core/Domain/Services/ModelConstructor.cs

```diff
@@ -112,6 +113,20 @@ private void finalizeModel(IModel model)
 
          //now we should be able to resolve all references
          _referencesResolver.ResolveReferencesIn(model);
+
+
+         //This should be done after reference were resolved to ensure that we do not remove formula parameter that could not be evaluated
+         removeUndefinedInstanceParametersIn(model);
```

Comment:
Once everything is created, I am removing explicitly parameter instances with NaN values